src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -334,6 +310,162 @@ public CaseDTO getCaseByCaseReference(final long caseRef, CaseQueryRequestDTO re
+    newAddress.setOrganisationName(modifyRequestDTO.getCeOrgName());
+
+    AddressModification payload =
+        AddressModification.builder()
+            .collectionCase(collectionCase)
+            .originalAddress(originalAddress)
+            .newAddress(newAddress)
+            .build();
+    sendEvent(EventType.ADDRESS_MODIFIED, payload, originalCaseId);
+  }
+
+  private void sendAddressTypeChangedEvent(
+      UUID updatedCaseId, UUID originalCaseId, ModifyCaseRequestDTO modifyRequestDTO) {
+    CollectionCase collectionCase = new CollectionCase();
+    collectionCase.setId(originalCaseId.toString());
+    collectionCase.setCeExpectedCapacity(modifyRequestDTO.getCeUsualResidents());
+    collectionCase.setContact(null);
+
+    Address address = new Address();
+    address.setAddressLine1(modifyRequestDTO.getAddressLine1());
+    address.setAddressLine2(modifyRequestDTO.getAddressLine2());
+    address.setAddressLine3(modifyRequestDTO.getAddressLine3());
+    address.setEstabType(modifyRequestDTO.getEstabType().getCode());
+    address.setOrganisationName(modifyRequestDTO.getCeOrgName());
+    address.setAddressType(modifyRequestDTO.getCaseType().name());
+
+    collectionCase.setAddress(address);
+
+    AddressTypeChanged payload =
+        AddressTypeChanged.builder()
+            .newCaseId(updatedCaseId)
+            .collectionCase(collectionCase)
+            .build();
+    sendEvent(EventType.ADDRESS_TYPE_CHANGED, payload, updatedCaseId);
+  }
+
+  private void updateModifiedResponse(CaseDTO response, ModifyCaseRequestDTO modifyRequestDTO) {
+    CaseType caseType = modifyRequestDTO.getCaseType();
+    response.setCaseType(caseType.name());
+    response.setAddressType(caseType.name());
+    EstabType estabType = modifyRequestDTO.getEstabType();
+    response.setEstabType(estabType);
+    response.setEstabDescription(estabType.getCode());
+    response.setAddressLine1(modifyRequestDTO.getAddressLine1());
+    response.setAddressLine2(modifyRequestDTO.getAddressLine2());
+    response.setAddressLine3(modifyRequestDTO.getAddressLine3());
+    response.setCeOrgName(modifyRequestDTO.getCeOrgName());
+  }
+
+  @Override
+  public CaseDTO modifyCase(ModifyCaseRequestDTO modifyRequestDTO) throws CTPException {
+    validateCompatibleEstabAndCaseType(
+        modifyRequestDTO.getCaseType(), modifyRequestDTO.getEstabType());
+    UUID originalCaseId = modifyRequestDTO.getCaseId();
+    UUID updatedCaseId = originalCaseId;
+
+    CaseContainerDTO caseDetails = retrieveCaseById(originalCaseId, false);
+    rejectHouseholdIndividual(caseDetails);
+    CaseType requestedCaseType = modifyRequestDTO.getCaseType();
+
+    boolean addressTypeChanged = isAddressTypeChange(requestedCaseType, caseDetails.getEstabType());
+
+    CaseDTO response = caseDTOMapper.map(caseDetails, CaseDTO.class);
+
+    if (addressTypeChanged) {
+      rejectNorthernIrelandHouseholdToCE(requestedCaseType, caseDetails);
+      updatedCaseId = UUID.randomUUID();
```

Comment:
newCaseId would be more appropriate

Reply:
I have changed this to "caseId" since if can be either the original or the new one.